### PR TITLE
[develop < T0100-GA] Wait for Active Connection

### PR DESCRIPTION
**Pull request by BorisTasevski**
### Description

Enable a wait for active connection

### Pull request type

Please delete options that are not relevant.

- [X] Feature
- [X] Code style update (formatting, renaming)
- [x] Refactoring without functional or API changes

### Related issues

[Have a way to wait for active connection](https://github.com/memgraph/gqlalchemy/issues/133)

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

######################################

### Reviewer checklist (the reviewer checks this part)
- [ ] Core feature implementation
- [ ] Tests
- [ ] Code documentation
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)

######################################

### CODE REVIEW

**Review by katarinasupe — CHANGES_REQUESTED** [review]

File: gqlalchemy/exceptions.py [hunk 1]
```diff
@@ -119,3 +123,23 @@ class GQLAlchemyExtraKeywordArgumentsInWhere(GQLAlchemyError):
     def __init__(self):
         super().__init__()
         self.message = EXTRA_KEYWORD_ARGUMENTS_IN_WHERE
+
+
+class GQLAlchemyConnectionError(GQLAlchemyError):
+    def __init__(self):
+        super().__init__()
```
Comment (katarinasupe) [inline gqlalchemy/exceptions.py:130]: Is this line necessary?

File: gqlalchemy/instance_runner.py [hunk 2]
```diff
@@ -90,27 +92,40 @@ def wait_for_docker_container(
...
         if time.perf_counter() - start_time >= timeout:
-            raise TimeoutError(DOCKER_TIMEOUT_ERROR_MESSAGE)
+            raise GQLAlchemyDockerTimeoutError
 
         container.reload()
         delay *= backoff
 
 
 def wait_for_memgraph(memgraph: Memgraph, delay: float = 0.01, timeout: int = 5, backoff: int = 2) -> None:
+    """Wait for a memgraph instance to become available.
```
Comment (katarinasupe) [inline gqlalchemy/instance_runner.py:112]: Memgraph*

File: gqlalchemy/instance_runner.py [hunk 3]
```diff
@@ -90,27 +92,40 @@ def wait_for_docker_container(
...
+
+    Args:
+        memgraph: The instance of memgraph in to be waited to become available.
```
Comment (katarinasupe) [inline gqlalchemy/instance_runner.py:115]: Memgraph*

File: gqlalchemy/instance_runner.py [hunk 4]
```diff
@@ -90,27 +92,40 @@ def wait_for_docker_container(
...
+        delay: A float that defines how long to wait between retries.
+        timeout: A float that defines how long to wait for the status.
+        backoff: An integer that defines the factor of multiplication of the delay.
+
+    Raises:
+      GQLAlchemyConnectionError: Raises an error when the memgraph instance isn't answering after the
```
Comment (katarinasupe) [inline gqlalchemy/instance_runner.py:121]: Memgraph*

**Review by katarinasupe — DISMISSED** [review]

File: gqlalchemy/instance_runner.py (hunk 2 above)
Comment (katarinasupe) [reply gqlalchemy/instance_runner.py:112]: @BorisTasevski did you fix this?

Comment (BorisTasevski) [reply gqlalchemy/instance_runner.py:112]: Oui, below as well

**Review by g-despot — APPROVED** [review]

**Review by Josipmrden — COMMENTED** [review]
Small changes needed

**Review by katarinasupe — APPROVED** [review]

**Pull request merged**